Fixes for traefik v3 (#872)
traefik/traefik-helm-chart: README.md

This chart bootstraps Traefik version 2 as a Kubernetes ingress controller,
using Custom Resources `IngressRoute`: <https://docs.traefik.io/providers/kubernetes-crd/>.

> Notice if you use traefik v3, crds with namespace traefik.containo.us were removed, and only traefik.io is available.

Reviewer: ```suggestion
It's now possible to use this chart with Traefik v3 beta3. See [Migration guide from v2 to v3](https://doc.traefik.io/traefik/v3.0/migration/v2-to-v3/) and upgrading section of this chart on CRDs move from `traefik.containo.us` to `traefik.io`
```

> It's now possible to use this chart with Traefik v3 beta3. See [Migration guide from v2 to v3](https://doc.traefik.io/traefik/v3.0/migration/v2-to-v3/) and upgrading section of this chart on CRDs move from `traefik.containo.us` to `traefik.io`

Reviewer: ```suggestion
It's now possible to use this chart with Traefik v3 beta3. See [Migration guide from v2 to v3](https://doc.traefik.io/traefik/v3.0/migration/v2-to-v3/) and upgrading section of this chart on CRDs.
```

### Upgrading CRDs

🛂 **Warning**: Traefik v3 totally removes the crd support for traefik.containo.us CRDs. By default this helm installs the CRDs compatible with v2 also, but Traefik v3 will no longer monitor them. There is no support for deprecation errors, so your existing resources may silently fail to work after upgrade to Traefik v3. [Migration guide from v2 to v3](https://doc.traefik.io/traefik/v3.0/migration/v2-to-v3/)

With Helm v3, CRDs created by this chart can not be updated, cf the [Helm Documentation on CRDs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions). Please read carefully release notes of this chart before upgrading CRDs.

```bash

> kubectl apply --server-side --force-conflicts -f https://github.com/traefik/traefik-helm-chart/traefik/crds/

Reviewer: It was a "-k" on purpose, to apply `kustomization.yaml`.
Either you can remove `kustomization.yaml` or update it and keep `-k`.

Author: Reverted

Author: Also forgot the new crd in that file

### Upgrading CRDs

> 🛂 **Warning**: Traefik v3 totally removes the crd support for traefik.containo.us CRDs. By default this helm installs the CRDs compatible with v2 also, but Traefik v3 will no longer monitor them. There is no support for deprecation errors, so your existing resources may silently fail to work after upgrade to Traefik v3. [Migration guide from v2 to v3](https://doc.traefik.io/traefik/v3.0/migration/v2-to-v3/)

Reviewer: ```suggestion
🛂 **Warning**: Traefik v3 totally removes the crd support for traefik.containo.us CRDs. By default this helm installs the CRDs compatible with v2 also, but Traefik v3 will no longer monitor them. There is no support for deprecation errors, so your existing resources may silently fail to work after upgrade to Traefik v3. See [Migration guide from v2 to v3](https://doc.traefik.io/traefik/v3.0/migration/v2-to-v3/) for more details.
```

With Helm v3, CRDs created by this chart can not be updated, cf the [Helm Documentation on CRDs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions). Please read carefully release notes of this chart before upgrading CRDs.

```bash
kubectl apply --server-side --force-conflicts -f https://github.com/traefik/traefik-helm-chart/traefik/crds/
```

for traefik v3, please use
```bash

> kubectl apply --server-side --force-conflicts -f https://raw.githubusercontent.com/traefik/traefik/v3.0/docs/content/reference/dynamic-configuration/kubernetes-crd-definition-v1.yml

Reviewer: I am not sure it's a good idea.  
It means that CRDs installed with the chart can be different when user upgrade. 
I understand you are trying to provide a way to install only Traefik v3 CRDs, which is a good idea, but it should provide consistent behavior between fresh install and upgrade.

Author: Not sure what to do about that since they decided to cut the CRDs in half, and even add an extra one for v3.
Leaving the containo.us as a valid CRD in the system without traefik monitoring it anymore seems like a good way to have people extremely confused about why nothing works anymore, and no errors if they copy some resource from some other place and dont immediately get an error

Author: I changed the wording some, and refered people to the migration guide

Reviewer: Considering how we can handle CRDs with Helm... it seems like we should fork and provide a `traefik3-helm-chart`... :thinking:

Author: I think it is more an issue that belongs upstream rather than for the helm. I do think that a banner on https://github.com/traefik/traefik-helm-chart/issues about the issue should prevent too many bug reports on the helm.
